load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java

```diff
@@ -15,122 +15,224 @@
+  )
+  private String project = "";
+  @Parameter(
+      names = {"--batch_size", "-b"},
+      description = "Number of messages to batch per publish request.",
+      validateWith = GreaterThanZeroValidator.class
+  )
+  private int batchSize = 10;
+  @Parameter(
+      names = {"--subscriber_fanout"},
+      description = "Number of subscription ids to use for each topic. Must be at least 1.",
+      validateWith = GreaterThanZeroValidator.class
+  )
+  private int subscriberFanout = 1;
+  @Parameter(
+      names = {"--broker"},
+      description = "The network address of the Kafka broker."
+  )
+  private String broker;
+  @Parameter(
+      names = {"--request_rate"},
+      description = "The rate at which each client will make requests."
+  )
+  private int requestRate = 10;
+  @Parameter(
+      names = {"--max_concurrent_requests"},
+      description = "The maximum number of concurrent requests each client will allow."
+  )
+  private int maxConcurrentRequests = 20;
+  @Parameter(
+      names = {"--burn_in_duration_seconds"},
+      description = "The duration, in seconds, to run without recording statistics in order to allow tuning."
+  )
+  private int burnInDurationSeconds = 20;
+  @Parameter(
+      names = {"--number_of_messages"},
+      description = "The number of messages to publish in the test. Enabling this will override " +
+          "--loadtest_length_seconds. Enabling this flag will also enable the check for message loss. If set less " +
+          "than 1, this flag is ignored."
+  )
+  private int numberOfMessages = 0;
 
   public static void main(String[] args) {
     // Turns off all java.util.logging.
     LogManager.getLogManager().reset();
+    Driver driver = new Driver();
+    JCommander jCommander = new JCommander(driver, args);
+    if (driver.help) {
+      jCommander.usage();
+      return;
+    }
+    driver.run();
+  }
+
+  private void run() {
     try {
-      // Parse command line arguments.
-      BaseArguments baseArgs = new BaseArguments();
-      CPSArguments cpsArgs = new CPSArguments();
-      KafkaArguments kafkaArgs = new KafkaArguments();
-      CompareArguments dataComparisonArgs = new CompareArguments();
-      JCommander jCommander = new JCommander(baseArgs);
-      jCommander.addCommand(CPSArguments.COMMAND, cpsArgs);
-      jCommander.addCommand(KafkaArguments.COMMAND, kafkaArgs);
-      jCommander.addCommand(CompareArguments.COMMAND, dataComparisonArgs);
-      jCommander.parse(args);
-      if (jCommander.getParsedCommand() == null) {
-        if (baseArgs.isHelp()) {
-          jCommander.usage();
-        }
-        return;
-      }
-      if (jCommander.getParsedCommand().equals(CompareArguments.COMMAND)) {
-        // Compares data dumps for correctness.
-        Comparison c = new Comparison(dataComparisonArgs.getFile1(), dataComparisonArgs.getFile2());
-        c.compare();
-        return;
+      Map<String, Map<ClientParams, Integer>> clientTypes = ImmutableMap.of(
```

Comment:
Just build up a Map\<ClientParams, Integer\> clientParamsMap first and then pass a:

ImmutableMap.of( "us-central1-a", clientParamsMap)

at line 177 below.